android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/HomeScreen.kt

```diff
@@ -19,15 +16,22 @@ import co.nimblehq.ic.kmm.suv.android.util.LoadingParameterProvider
 import org.koin.androidx.compose.getViewModel
 
 @Composable
-fun HomeScreen(viewModel: HomeViewModel = getViewModel()) {
+fun HomeScreen(
+    viewModel: HomeViewModel = getViewModel(),
+    onSurveyDetailClick: (SurveyArgument) -> Unit = {}
+) {
     val currentDate by viewModel.currentDate.collectAsState()
     val avatarUrl by viewModel.avatarUrlString.collectAsState()
     val isLoading by viewModel.isLoading.collectAsState()
     val errorMessage by viewModel.errorMessage.collectAsState()
     val surveysUiModel by viewModel.surveysUiModel.collectAsState()
+    var isLaunched by remember { mutableStateOf(false) }
 
     LaunchedEffect(Unit) {
-        viewModel.loadProfileAndSurveys()
+        if (!isLaunched) {
+            viewModel.loadProfileAndSurveys()
+            isLaunched = true
+        }
```

Comment:
Does this update to prevent re-fetching data on navigating back from the details screen? 🤔 If so, I think we could add the `isLaunched` flag as the key of `LaunchedEffect` instead of `Unit` and get rid of the `if (!isLaunched)` condition.

Reply:
for the key of `LaunchedEffect`, the launched effect will call again if the value is changed. If we use the `isLaunched` flag, it will call 2 times.

I just added TODO into the LaunchedEffect block. I will check this later. I just found that my approach also did not prevent re-fetching data :D

Reply:
@suho 🆗 Deal 🤝